addon/components/imgix-image.js

```diff
@@ -192,50 +159,130 @@ export default Component.extend({
         );
       }
 
+      // Setup
+      const widthProp = get(this, 'width');
+      const heightProp = get(this, 'height');
+      const pathAsUri = get(this, '_pathAsUri');
+      const disableSrcSet = get(this, 'disableSrcSet');
+      const client = get(this, '_client');
+      const buildWithOptions = options =>
+        client.buildURL(pathAsUri.path(), options);
+
+      const isFixedDimensionsMode = widthProp != null || heightProp != null;
+
+      const shouldShowDebugParams = get(config, 'APP.imgix.debug');
+
+      const { imgixOptions, aspectRatio } = (() => {
+        let imgixOptions = {
+          ...get(this, 'options')
+        };
+        let aspectRatio = get(this, 'aspectRatio');
+        if (imgixOptions.ar) {
+          aspectRatio = imgixOptions.ar;
+          delete imgixOptions.ar;
+        }
+
+        return { imgixOptions, aspectRatio };
+      })();
+
+      // Calculate AR
+      const aspectRatioDecimal = parseAspectRatio(aspectRatio);
+      if (aspectRatio != null && aspectRatioDecimal === false) {
+        // false return value from parseAspectRatio indicates invalid response
+        // eslint-disable-next-line no-console
+        console.warn(
+          `[imgix] The aspect ratio passed ("${aspectRatio}") is not in the correct format. The correct format is "W:H".`
+        );
+      }
+
+      // Ensure width and height are set correctly according to aspect ratio
+      const { width, height } = (() => {
+        if (widthProp && heightProp && aspectRatio) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `[imgix] All three of width, height, and aspect ratio were passed. The aspect ratio prop has not effect in this configuration.`
+          );
+        }
+        if (!widthProp && !heightProp && disableSrcSet && aspectRatio) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `[imgix] The aspect ratio prop has not effect when both width and height are not set, and when srcsets are disabled. To use aspect ratio, please either pass width or height values, or enable src sets.`
+          );
+        }
+
+        if (
+          !(widthProp || heightProp) ||
+          (widthProp && heightProp) ||
+          !aspectRatioDecimal
+        ) {
+          return { width: widthProp, height: heightProp };
```

Comment:
I'm mildly concerned that this line here allows the user to pass `w: 0` to the URL builder. Should we do some coercion at some point to make sure that zeroes passed to either dimension will get cast to `null` or `undefined` and therefore left off the final URL altogether?

Reply:
With these kind of things, I often find it gets to a stage of "where do we stop (implementing checks)". Although people can pass w=0 or h=0 to the url builder, I think it would be easy for them to debug their application to see that they're passing w=0 or h=0, and so I'm hesitant to implement some sort of logic that coerces the value into a "nicer" value for our implementation, as it makes it more of a black box.

I'd be happy to implement a warning/error though, say if they passed either a width or a height that wasn't in the range (0,∞) and also wasn't a number.

Reply:
Let's go with a warning, then. Feel free to just open up a ticket for this if you don't want to include it in this PR.

Reply:
Added it here 👍